consensus/src/chained_bft/consensus_types/vote_msg.rs

```diff
@@ -78,32 +75,8 @@ impl VoteMsg {
         }
     }
 
-    /// All the vote data getters are just proxies for retrieving the values from the VoteData
-    pub fn block_id(&self) -> HashValue {
-        self.vote_data.block_id()
-    }
-
-    pub fn executed_state_id(&self) -> HashValue {
-        self.vote_data.executed_state_id()
-    }
-
-    pub fn block_round(&self) -> Round {
-        self.vote_data.block_round()
-    }
-
-    pub fn parent_block_id(&self) -> HashValue {
-        self.vote_data.parent_block_id()
-    }
-
-    pub fn parent_block_round(&self) -> Round {
-        self.vote_data.parent_block_round()
-    }
-
-    pub fn grandparent_block_id(&self) -> HashValue {
-        self.vote_data.grandparent_block_id()
-    }
-    pub fn grandparent_block_round(&self) -> Round {
-        self.vote_data.grandparent_block_round()
+    pub fn vote_data(&self) -> &VoteData {
```

Comment:
Lol ok this is a matter of taste. We actually hide `VoteData` on purpose. As far as I remember @zekun000 wanted `VoteData` to be purely internal implementation detail of `VoteMsg`. I'm ok with both options, will let you guys decide.

Reply:
As a general comment, I'm concerned about the large amount of calls with 7+ arguments in consensus:
https://gist.github.com/ec8d79f88cbcbacde5e3a21a6b207358

Reply:
so the next steps are replacing those arguments by a single `vote_data()`?

Reply:
To have a `VoteMsg` constructor that takes a `VoteData`, yeah.